### DISCUSSION TITLE
Option to cancel the update

### DISCUSSION THREAD

**Discussion by priechodsky** [Romanitho/Winget-AutoUpdate#104] [Ideas]
In the popup, the message how many SW updates are available for e.g. 10 seconds and the ability to cross the entire script to stop currently, while it would run again, only after re-login the next day.

**Comment by Romanitho** [comment]
It is not possible, as the task run under system and the notification is displayed to user. No way to interrupt a running system script  with users rights

**Reply by KnifMelti** [reply]
Maybe now with the latest implementations when it comes to user accessing the tasks...

**Reply by KnifMelti** [reply]
> Maybe now with the latest implementations when it comes to user accessing the tasks...

Had a solution, but toasts handling in PS 5 requires more skills than I have (@Romanitho ) to implement a custom protocol handle for running a script in a ToastOnClickAction...
...a common dialog in winget-upgrade.ps1 ending the task does the trick, but it's not as nice...

**Reply by Romanitho** [reply]
We can maybe implement a pop-up (or notification) that says how many updates are available and if user is ok to install them. But when process starts, not easy to stop.

**Reply by ghost** [reply]
This would be very nice! That would be absolutely enough, but the user definitely needs the control. it can simply be that the timing doesn't fit. that's what has always annoyed our users about intune.
